src/plugins/vis_types/timeseries/public/application/components/lib/convert_series_to_datatable.ts

```diff
@@ -151,15 +164,22 @@ export const convertSeriesToDataTable = async (
     const filtersColumn = columns.find((col) => col.type === BUCKET_TYPES.FILTERS);
     let rows: DatatableRow[] = [];
     for (let j = 0; j < seriesPerLayer.length; j++) {
-      const data = seriesPerLayer[j].data.map((rowData) => {
-        const row: DatatableRow = [rowData[0], rowData[1]];
-        // If the layer is split by terms aggregation, the data array should also contain the split value.
+      const { data, label, isSplitByTerms, termsSplitKey } = seriesPerLayer[j];
+      const seriesData = data.map((rowData) => {
+        let rowId = X_ACCESSOR_INDEX;
+        const rowsData = { [rowId++]: rowData[0], [rowId++]: rowData[1] };
+
+        let splitValue;
         if (isGroupedByTerms || filtersColumn) {
-          row.push([seriesPerLayer[j].label].flat()[0]);
+          const termsValue = Array.isArray(termsSplitKey)
+            ? new MultiFieldKey({ key: termsSplitKey })
+            : termsSplitKey;
+          splitValue = { [rowId]: isSplitByTerms && termsValue ? termsValue : [label].flat()[0] };
```

Comment:
@DziyanaDzeraviankina actually I found one bug (happens also in main but lets fix it here!)
When you split by a boolea field, the false value doesnt trigger a filter.
To fix it, I think we have to change this row to:

```
splitValue = { [rowId]: isSplitByTerms && termsValue !== undefined ? termsValue : [label].flat()[0] };
```

Reply:
Thanks for noticing!